lib/package.gi

```diff
@@ -2203,9 +2203,40 @@ InstallGlobalFunction( ValidatePackageInfo, function( info )
         x -> IsString(x) and 0 < Length(x) and x[1] <> '=',
         "a nonempty string that does not start with `='" );
     TestMandat( record, "Date",
-        x -> IsString(x) and Length(x) = 10 and x{ [3,6] } = "//"
-                 and ForAll( x{ [1,2,4,5,7,8,9,10] }, IsDigitChar ),
-        "a string of the form `dd/mm/yyyy'" );
+        x -> IsString(x) and Length(x) = 10
+          and ( ( x{ [3,6] } = "//" and ForAll( x{ [1,2,4,5,7,8,9,10] }, IsDigitChar ) )
+          or ( x{ [5,8] } = "--" and ForAll( x{ [1,2,3,4,6,7,9,10] }, IsDigitChar ) ) ) ,
+        "a string of the form `yyyy-mm-dd` or `dd/mm/yyyy`" );
+
+    #If the date is in the format `dd/mm/yyyy` a message is printed
+    if IsBound( record.Date ) and  record.Date{ [3,6] } = "//" then
+       Info( InfoPackageLoading, 2, Concatenation(record.PackageName, ": Please be adviced to change the date format to `yyyy-mm-dd`"));
+    fi;
+
+    #check if the date is valid
+    CheckDateValidity := function(x)
```

Comment:
Wouldn't it be better to parse the date into components `dd`, `mm` and `yyyy`, and then do a single check instead of duplicating those lines dependently on the list layout?

Reply:
Yes that's better. I'd suggest the following which only makes the distinction for the split
```suggestion
    #check if the date is valid
    CheckDateValidity := function(x)
      local date;
      if x{ [3,6] } = "//" then #for the old format split at '/'
        date := List( SplitString( x, "/" ), Int);
      elif x{ [5,8] } = "--" then #for the yyyy-mm-dd format split at '-'
        date := List( SplitString( date, "-" ), Int);
        Permuted(date, (1,3)); #sort such that date=[dd,mm,yyyy]
      fi;
      if IsBound(date) and (not (date[1] in [1..31] and
                   date[2] in [1..12] and
                   date[3] > 1999 and # GAP 4 appeared in 1999
                   date[1] <= DaysInMonth( date[2], date[3] ))) then
              return false;
      fi;
    return true;
    end;
```

Reply:
If that way is okay, I'll also change this in the PR  [#78](https://github.com/gap-system/gap-distribution/pull/78) in gap-distribution.

Reply:
looks sensible to me

Reply:
you might want to add an `else Error("date must be in form yyyy-mm-dd or dd/mm/yyyy");` or so

Reply:
As far as I understand it this is checked a couple of lines above:
```
TestMandat( record, "Date",
        x -> IsString(x) and Length(x) = 10
          and ( ( x{ [3,6] } = "//" and ForAll( x{ [1,2,4,5,7,8,9,10] }, IsDigitChar ) )
          or ( x{ [5,8] } = "--" and ForAll( x{ [1,2,3,4,6,7,9,10] }, IsDigitChar ) ) ) ,
        "a string of the form `yyyy-mm-dd` or `dd/mm/yyyy`" );
```

Reply:
Yes, but what if the test above passed, so no `yyyy-mm-dd` or `dd/mm/yyyy` suggestion was displayed?

The code "which only makes the distinction for the split" looks good to me, please update the PR.